ens/main.py

```diff
@@ -52,7 +52,7 @@
         BaseProvider,
     )
     from web3.types import (  # noqa: F401
```

Comment:
I think it would be better to have `web3.types` use the `if TYPECHECKING` pattern and allow direct imports from this module...

Reply:
Agreed, but it seems to me like it might be necessary in the `ens` module, since the docs build breaks when trying to import `ENS` and I haven't been able to diagnose why. In `web3` we can directly import from `web3.types`